timesketch/lib/analyzers/feature_extraction.py

```diff
@@ -146,6 +178,57 @@ def run(self):
 
         return ', '.join(return_strings)
 
+    @staticmethod
+    def _get_attribute_value(
+            current_val,
+            extracted_value,
+            keep_multi,
+            merge_values,
+            type_list):
+        """Return value to store.
+        Args:
+            current_val: current value of store_as.
+            extracted_value: values matched from regexp (type list).
+            keep_multi: choise if you keep all match from regex (type boolean).
+            merge_values: choise if you merge value from extracted
+                 and current (type boolean).
+            type_list: choise if you store values in list type(type boolean).
+        Returns:
+            Value to store
+        """
+        new_value = None
+        if not current_val:
+            merge_values = False
+        if len(extracted_value) == 1:
+            keep_multi = False
+        if type_list:
+            if merge_values:
```

Comment:
I would opt here

```
if merge_values and keep_multi:
    return list(set(current_val) | ....

if merge_values and extracted_value[0] not in current_val:
  current_val.append(extracted_value[0])
  return current_val

```

Reply:
that is, this code has a lot of if ... if else... if ... if elif ... and deep in.. .where you can save a lot of that by returning early and then skipping the else


also can you add unit tests for this function? Test it with few scenarios to make sure it is working as expected.